### DISCUSSION TITLE
support for python 3.7.x

### DISCUSSION THREAD

**Discussion by amendoza28** [endangeredoxen/fivecentplots#51] [General]
is there plan to support python 3.7?

when I installed 3.7 in my computer, I got this error "fatal python error take_gil null tstate" when I was trying to import fivecentplots library

**Comment by endangeredoxen** [comment]
I have test set to run for py 3.6, 3.8, and 3.10 (which all pass) so I'm not sure why 3.7 would be an odd ball.  I'll take a look and see if I can replicate your error.  A quick google search shows people complaining about this on a 3.6 distribution (without fcp) and I know fcp works there.  Might be completely unrelated to fcp...

**Comment by endangeredoxen** [comment]
OK, on an Ubuntu machine I created a fresh python 3.7 environment with conda and installed fcp:

`conda create -n "py37" python=3.7 -y && conda activate py37 && pip install fivecentplots`

Then I ran one of the basic fcp [examples](https://endangeredoxen.github.io/fivecentplots/0.5.2/api/plot.html):

```
import fivecentplots as fcp
from pathlib import Path
import pandas as pd
df = pd.read_csv(Path(fcp.__file__).parent / 'test_data' / 'fake_data.csv')
fcp.plot(df, x='Voltage', y='I [A]', legend=['Die', 'Substrate'], ax_size=[400, 300], filter='Target Wavelength==450 & Temperature [C]==25 & Boost Level==0.2')
```

Everything worked fine.  I will mention that when I tried to create a 3.7 environment on my Mac I was told this version wasn't supported with my hardware.  There appears to be a hack to get it working but I'm not going to explore that.  I suspect the problem is specific to your current configuration.

**Reply by amendoza28** [reply]
Thank you for checking this, I went back to 3.6 version, and now it works fine, I will probably need to switch back to 3.7 due to other software. I am using the fivecentplots library in Windows10, I will install this library in another system, and check again, I let you know. thank you!
